pkg/pool-hooks/contracts/MevHook.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IHooks } from "@balancer-labs/v3-interfaces/contracts/vault/IHooks.sol";
+import { IMevHook } from "@balancer-labs/v3-interfaces/contracts/pool-hooks/IMevHook.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import {
+    HooksConfig,
+    HookFlags,
+    LiquidityManagement,
+    PoolSwapParams,
+    TokenConfig,
+    MAX_FEE_PERCENTAGE
+} from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+
+import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
+import { VaultGuard } from "@balancer-labs/v3-vault/contracts/VaultGuard.sol";
+import { BaseHooks } from "@balancer-labs/v3-vault/contracts/BaseHooks.sol";
+
+contract MevHook is BaseHooks, SingletonAuthentication, VaultGuard, IMevHook {
+    using FixedPoint for uint256;
+
+    // Max Fee is 99.9999% (Max supported fee by the vault).
+    uint256 private constant _MEV_MAX_FEE_PERCENTAGE = MAX_FEE_PERCENTAGE;
+
+    bool internal _mevTaxEnabled = false;
+    uint256 internal _defaultMevTaxMultiplier = 0;
+    uint256 internal _defaultMevTaxThreshold = 0;
+    mapping(address => uint256) internal _poolMevTaxMultipliers;
+    mapping(address => uint256) internal _poolMevTaxThresholds;
+
+    modifier withRegisteredPool(address pool) {
+        HooksConfig memory hooksConfig = _vault.getHooksConfig(pool);
+
+        if (hooksConfig.hooksContract != address(this)) {
+            revert MevHookNotRegisteredInPool(pool);
+        }
+
+        _;
+    }
+
+    constructor(IVault vault) SingletonAuthentication(vault) VaultGuard(vault) {
+        // solhint-disable-previous-line no-empty-blocks
+    }
+
+    /// @inheritdoc IHooks
+    function onRegister(
+        address,
+        address pool,
+        TokenConfig[] memory,
+        LiquidityManagement calldata
```

Comment:
We should validate that `onCallDynamic...` is set to true; otherwise this doesn't work right?

Reply:
ah this is set here in the getter; nvm

Reply:
Yes, exactly, any pool will be able to use this, we don't want to restrict the factory. I'll check the disable of unbalanced liquidity.